src/lib.rs

```diff
@@ -88,42 +90,167 @@ impl InlinerBuilder {
         };
         let result =
             Visitor::<R>::with_resolver(src_file, self.root, errors.as_mut(), Cow::Owned(resolver))
-                .visit();
+                .visit()
+                .map_err(|kind| Error::Initial(kind))?;
         match errors {
             Some(ref errors) if errors.is_empty() => Ok(result),
             None => Ok(result),
-            Some(errors) => Err(Error::NotFound(errors)),
+            Some(errors) => Err(Error::Inline(errors)),
         }
     }
 }
 
-/// A source location
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct SourceLocation {
-    pub path: PathBuf,
-    pub line: usize,
-    pub column: usize,
-    _private: (),
+/// An error that was encountered while reading, parsing or inlining a module.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum Error {
+    /// An error happened while reading or parsing the initial file.
+    ///
+    /// For example, the initial file wasn't found or wasn't a valid Rust file.
+    Initial(ErrorKind),
+
+    /// The initial file was successfully loaded, but some errors happened while attempting to
+    /// inline modules.
+    Inline(Vec<InlineError>),
```

Comment:
Based on [this conversation](https://github.com/TedDriggs/syn-inline-mod/pull/12/files#r451575863), I wonder if this should be `Error::Incomplete(IncompleteInlining)` where `IncompleteInlining` is:

```rust
#[non_exhaustive]
pub struct IncompleteInlining {
    pub file: syn::File,
    pub errors: Vec<InlineError>,
}
```

That way a caller can still get whatever succeeded _and_ have the errors.